riscv-plic.adoc

```diff
@@ -17,102 +17,436 @@ https://creativecommons.org/licenses/by/4.0/.
+
+|Length in Bits of each Interrupt Source Priority
+|2
+|21 + n
+|Length in bits of interrupt source priority.
+
+|Starting Offset to Interrupt Pending Bits Block
+|4
+|23 + n
+|The relative offset to PLIC physical address which points to interrupt pending block. Value of zero means no interrupt pending bits supported in PLIC core.
+|Number of Hart Context Interrupt Description Structures
+|1
+|27 + n
+|Number of Hart context interrupt structures follow PLIC structure. See *Table 5-68*.
+
+|Hart Context Interrupt Description (HCID) Structures
+|n * x
+|28 + n
+|The first HCID structure. Total length in byte for each HCID is referred as “x”.
+|===
+
+*Table 5-68 PLIC HCID Structure*
+[cols="35%,10%,10%,45%"]
+|===
+| *Field* | *Byte Length*|*Byte Offset *| *Description*
+|Hart ID
+|1
+|0
+|ID of Hart owns these interrupt sources. The value specified in this field must be one of value in PLIC Target Hart ID [n] in *Table 5-67* PLIC structure.
+
+|Privilege Level
+|1
+|1
+|The privilege levels of this Hart. +
+0: User Mode +
+1: Supervisor Mode +
+2: Reserved +
+3: Machine Mode
+
+|Starting Offset to Interrupt Enable Bits Block
+|4
+|2
+|The relative offset to PLIC physical address which points to interrupt enable bits block. Value of zero means no interrupt enable bits supported in PLIC. The interrupt enable bits block is used to enable specific interrupt source for the Hart specified in Hart ID and Privilege Mode specified in Privilege Level in this table (PLIC HCID Structure)
+
+|Offset to the Interrupt Priority Threshold
+|4
+|6
+|The relative offset to PLIC physical address which points to interrupt priority threshold of the Hart specified in Hart ID and Privilege Mode specified in Privilege Level in this table (PLIC HCID Structure). The valid value is in the range of Maximum Interrupt Priority Levels in *Table 5-67 PLIC structure*. The bit length of interrupt priority is specified in Length in Bits of each Interrupt Source Priority in Table *5-67 PLIC structure*.
+
+|Length in Byte of Interrupt Priority Threshold
+|10
+|1
+|The length of Interrupt Priority Threshold register of the Hart specified in Hart ID and Privilege Mode specified in Privilege Level in this table (PLIC HCID Structure)
+
+|Offset to Interrupt Claim/Complete
+|4
+|10
+|The relative offset to PLIC physical address which points to interrupt Claim/Complete register of the Hart specified in Hart ID and Privilege Mode specified in Privilege Level in this table (PLIC HCID Structure).
+|===
+
+
+image::https://github.com/changab/riscv-plic-acpi-images/blob/master/Figure5-24.jpg[GitHub][1000,705]
+
+#### Figure 5.24 PLIC-Global System Interrupts (Single Processor and Single PLIC Scenario)
+
+image::https://github.com/changab/riscv-plic-acpi-images/blob/master/Figure5-25.jpg[GitHub][1000,705]
+
+#### Figure 5.25 PLIC-Global System Interrupts (Multiple Processors and Multiple PLICs Scenario)
+
+
+## Device Tree Syntax for RISC-V PLIC
+*TBD*
```

Comment:
For reference, the current device tree syntax is:
```
        interrupt-controller@c000000 {
            #interrupt-cells = <0x00000001>;
            compatible = "riscv,plic0";
            interrupt-controller;
            interrupts-extended = <0x00000010 0x0000000b 0x00000011 0x0000000b 0x00000011 0x00000009 0x00000012 0x0000000b 0x00000012 0x00000009 0x00000013 0x0000000b 0x00000013 0x00000009 0x00000014 0x0000000b 0x00000014 0x00000009>;
            reg = <0x00000000 0x0c000000 0x00000000 0x04000000>;
            reg-names = "control";
            riscv,max-priority = <0x00000007>;
            riscv,ndev = <0x00000035>;
            phandle = <0x0000000c>;
        };
```

Reply:
Thanks, @fintelia
I just committed changes for Device Tree syntax